### PR TITLE
Feature: Users.recoverKeysByPassphrase and Users.backupKeysByPassphrase

### PR DESCRIPTION
**Pull request by perfectmak**
## Description

- This PR implements the `recoverKeysByPhassphrase` and `backupKeysByPassphrase` in the sdk.
- It also updates the documentation of the sdk with relevant examples.
- Minor update to how integration test configuration is being performed

Fixes: ch20496 & ch21007

## TODO
- [ ] Unable to complete integration testing until `getApiSessionToken()` is implemented, but the PR is ready for review now.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

- [x] Unit Test
- [x] Integration Test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by jsonsivar — APPROVED** [review]
LGTM. Thank you for the great documentation.

File: packages/users/src/vault/types.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * Object representing item stored or retrieved from the vault.
+ *
+ */
+export interface VaultItem {
+  itemType: VaultItemType;
+  value: string;
+}
+
+export enum VaultBackupType {
+  Password = 'password',
```
Comment (jsonsivar) [inline packages/users/src/vault/types.ts:19]: This won't be an option going forward but up to you if you want to clean it up here or not.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -40,6 +40,10 @@ jobs:
 
       - name: Integrations tests
         run: npm run test:integration
+        env:
+          HUB_AUTH_ENDPOINT: ${{ secrets.HUB_AUTH_ENDPOINT }}
```
[the remark this answers is not in the document]
Comment (perfectmak) [reply .github/workflows/test.yml:44]: Currently pointing to the dev environment for now 😅

File: packages/users/src/vault/types.ts (hunk 1 above)
Comment (perfectmak) [reply packages/users/src/vault/types.ts:19]: True, I'll clean it up.

**Pull request merged**